7.7.1/stretch/Dockerfile

```diff
@@ -0,0 +1,55 @@
+FROM debian:stretch-slim
+LABEL maintainer "Dave Curylo <dave@curylo.org>, Michael Hendricks <michael@ndrix.org>"
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    libarchive13 \
+    libgmp10 \
+    libossp-uuid16 \
+    libssl1.1 \
+    ca-certificates \
+    libdb5.3 \
+    libpcre3 \
+    libedit2 \
+    libgeos-c1v5 \
+    libspatialindex4v5 \
+    unixodbc \
+    odbc-postgresql \
+    tdsodbc \
+    libmariadbclient18 && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV SWIPL_VER 7.7.1
+ENV SWIPL_CHECKSUM fda2c8b6b606ff199ea8a6f019008aa8272b7c349cb9312ccd5944153509503a
+ENV BUILD_DEPS make gcc g++ wget libarchive-dev libgmp-dev libossp-uuid-dev libpcre3-dev \
+	       libreadline-dev libedit-dev libssl-dev zlib1g-dev libdb-dev libgeos++-dev \
+	       libspatialindex-dev unixodbc-dev git autoconf
+
+ENV SPACE_SHA1 cd6fefa63317a7a6effb61a1c5aee634ebe2ca05
+
+RUN apt-get update && apt-get install -y --no-install-recommends ${BUILD_DEPS} && \
+    mkdir /tmp/src && \
+    cd /tmp/src && \
+    wget http://www.swi-prolog.org/download/devel/src/swipl-${SWIPL_VER}.tar.gz && \
+    echo "${SWIPL_CHECKSUM}  swipl-${SWIPL_VER}.tar.gz" >> swipl-${SWIPL_VER}.tar.gz-CHECKSUM && \
+    sha256sum -c swipl-${SWIPL_VER}.tar.gz-CHECKSUM && \
+    tar -xzf swipl-${SWIPL_VER}.tar.gz && \
+    cd swipl-${SWIPL_VER} && \
+    cp build.templ build && \
+    sed -i '/PREFIX=$HOME/c\PREFIX=/swipl' build && \
+    sed -i '/# export DISABLE_PKGS/c\export DISABLE_PKGS="jpl xpce"' build && \
+    sed -i 's/# *\(.*--disable-libdirversion\)/\1/' build && \
+    chmod u+x build && ./build && \
+    rm -r /tmp/src && \
+    cd /usr/bin && ln -s /swipl/bin/swipl swipl && \
+    mkdir /swipl/lib/swipl/pack && \
+    cd /swipl/lib/swipl/pack && \
+    git clone https://github.com/JanWielemaker/space.git && \
```

Comment:
I'm against a `-space`.   We have SWI-Prolog's real core, its library that is bundled with the core, its packages that are bundled in the tar archive and can be compiled separately and community packages.   None of the boundaries are very hard.  `Space` went from the 2nd to the 3rd class because it is fairly hard to install in arbitrary environments, causing lots of porting issues while it is not vital.   I got feedback from two people telling me that including space was a good idea (they looked at the Dockerfile and saw it included), so I first wanted to make that work.

Supporting space in the docker image doesn't change much of its size, while in the controlled Debian stretch environment we can get it installed fairly easily.   The same goes for a couple of other add-ons, I think some 3 or 4.  They add seconds to the build time and altogether probably a couple of Mb to the image size.  Including them will make the image read-to-use for a significantly larger class of server tasks without having to setup the build essentials and add them to your own image.   I think there are two options:

  - Just include them
  - Have a `swipl:xl` version. 

Considering simplicity for the users I'm in favour of (1).  Note that we are only interested in mixed C/Prolog add-ons.  Pure Prolog ones are installed very quickly.

In particular when we have a layered Dockerfile the thing should be quite manageable.  I'll spend some time adding this stuff today.  Getting it to build is hard, while reorganizing is quick.